prime-dotnet-webapi/Controllers/SatEnrolmentController.cs

```diff
@@ -195,6 +199,7 @@ public async Task<ActionResult> SubmitSatEnrollee(int satId)
 
             var existsInPlr = await _plrProviderService.PartyExistsInPlrWithCollegeIdAndNameAsync(satId);
             var submission = await _partyService.CreateSubmissionAsync(satId, SubmissionType.SatEnrollee, existsInPlr);
+            await _keycloakClient.AssignRealmRole(User.GetPrimeUserId(), Roles.PhsaEformsSat);
```

Comment:
I would not use `User.GetPrimeUserId()` here.  Using `User` ties this to request state unnecessarily (making it harder to test) and could theoretically introduce a bug if we ever made this endpoint callable by admins (in that case, this would give the role to the _admin_, rather than the user).  In general, you should write your code to be defensive against possible bugs when it doesn't sacrifice anything.

I would use `satEnrollee.UserId`